### PR TITLE
connections/host/port/ws refactor + tests

### PR DESCRIPTION
**Pull request by mixmix**
The purpose of the PR is to close over some dangerous looseness in the connection setting behaviour.

When I say **"legacy config"** in this PR I mean `config.host` `config.port` `config.ws`, while **"new config"** is `connections.incoming`

## Possible pathways

1. **DEFAULT MODE** : don't add anything and you get happy new and old config provided
2. **LEGACY MODE** : set any of the old style config (`host` `port` `ws`)
  - default `connections.incoming` is generated based on what you provided
  - `host`, `port`, `ws.port` are then derived from the that `connections.incoming`
3. **NEW MODE** : set `connections.incoming`
  - `host`, `port`, `ws.port` are then derived from the that `connections.incoming`
4. **MIXED MODE** !! : set legacy things and `connections.incoming`
  - _just works_ ^:tm: 
  - if you `port` `host` `ws` lead to contradictions with your declared `connections.incoming` an error is thrown.

## The algorithm

1. see what building the config is based on opts RC
2. if no `connections.incoming` set that
  - based on anything in `config.host` `config.port` `config.ws` picked up along the way
  - fall back to some default (e.g. net port 8008)
3. check if the `connections.incoming` contradicts the the legacy config 
4. over-write the legacy config _based on the new config_
  - this ensures that legacy setup is always catered for, and also guards against divergence
---

**TODO**: 
- :heavy_check_mark:  write tests for all the above 
- :heavy_check_mark: link to `ssb-server` and see if its tests pass
- :heavy_check_mark: link to `patchbay` and see if it runs
- _ ask about my connections getters

### CODE REVIEW

File: util/get-net.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  return {
+    host: connection.host,
+    port: connection.port
+  }
+}
+
+function isAccessible (transport) {
+  return transport.scope === 'private' || transport.scope.includes('private') ||
+    transport.scope === 'public' || transport.scope.includes('public')
+}
```
Comment (mixmix) [inline util/get-net.js]: I'm not 100% sure how to choose which transportSetting within `connections.incoming.net` array we should be using for `config.host` and `config.port`

Here's I'm getting the first transportSetting which is .. accessible to the outside, which I think means a public or private scope.

**Comment by mixmix** [comment]
@dominictarr here's a juicy testing PR :watermelon:  :sweat_drops:

**Comment by dominictarr** [comment]
merged into 3.2.0

File: util/get-net.js (hunk 1 above)
Comment (mixmix) [reply util/get-net.js]: you didn't answer this question .. I hope it's fine @dominictarr

**Pull request merged**